mm2src/coins/eth.rs

```diff
@@ -778,9 +772,10 @@ async fn withdraw_impl(coin: EthCoin, req: WithdrawRequest) -> WithdrawResult {
     };
 
     let (tx_hash, tx_hex) = match coin.priv_key_policy {
-        EthPrivKeyPolicy::KeyPair(ref key_pair) => {
+        EthPrivKeyPolicy::Iguana(_) | EthPrivKeyPolicy::HDWallet { .. } => {
+            // Todo: nonce_lock is still global for all addresses but this needs to be per address
```

Comment:
I guess it's better to create an issue make a nonce_lock be accessible for each address.

Reply:
Todos are easier for devs inspecting the code if a problem arises from this. Also, this Todo is for me since I will fix this in upcoming PRs as we will require to use any address for swaps/orders without enabling/disabling coins which will require for this to be fixed.